File: sdks/node/src/api/processor/mapper.ts
import type { ProcessorRes } from "./res.js";
import type { ProcessorOutput } from "../../domain/processor/output.js";
import type { ProcessorReq } from "./req.js";
import type { ProcessorInput } from "../../domain/processor/input.js";
import { CyanMapper } from "../core/core_mapper.js";

class ProcessorMapper {
  public static toDomain(req: ProcessorReq): ProcessorInput {
    return {
      readDirectory: req.readDir,
      writeDirectory: req.writeDir,
      config: req.config,
      globs: req.globs.map((x) => CyanMapper.globReqToDomain(x)),
    } satisfies ProcessorInput;
  }

  public static toRes(res: ProcessorOutput): ProcessorRes {
    return {
      outputDir: res.directory,
    };
  }
}

export { ProcessorMapper };

File: sdks/python/cyan_sdk/api/processor/mapper.py
from cyan_sdk.api.core.core_mapper import CyanMapper
from cyan_sdk.api.processor.req import ProcessorReq
from cyan_sdk.api.processor.res import ProcessorRes
from cyan_sdk.domain.processor.input import ProcessorInput
from cyan_sdk.domain.processor.output import ProcessorOutput


class ProcessorMapper:
    @staticmethod
    def to_domain(req: ProcessorReq) -> ProcessorInput:
        globs = [CyanMapper.glob_req_to_domain(x) for x in req.globs]
        return ProcessorInput(
            read_directory=req.read_dir,
            write_directory=req.write_dir,
            config=req.config,
            globs=globs
        )

    @staticmethod
    def to_res(res: ProcessorOutput) -> ProcessorRes:
        return ProcessorRes(output_dir=res.directory)
